Orange/widgets/data/owpythonscript.py

```diff
@@ -513,25 +520,36 @@ def __init__(self):
         self.controlArea.layout().addStretch(1)
         self.resize(800, 600)
 
+    def _set_dict(self, dic, input, id):
+        id = id[0] if isinstance(id, dict) else id
+        if input is not None:
+            dic[id] = input
+        elif id in dic.keys():
+            del dic[id]
+
     @Inputs.in_data
-    def setExampleTable(self, et):
+    def setExampleTable(self, et, id=None):
         self.in_data = et
```

Comment:
This is not OK: `self.in_data` is going to contain the last received signal. If I recall correctly, we said that `in_data` is going to contain some data if exactly one data is present. If there are more, `in_data` should be `None`; it there are multiple inputs, the content of `in_data` would be picked essentially random and thus useless.

Perhaps the cheapest solution is to make it a property. Remove this line and `self.in_data = None`, and add

@property
def in_data(self):
    if len(self.in_datas) == 1:
        return next(self.in_datas.itervalues())

Or, cleaner, fix `initial_locals_state` to set `in_data` to `None` if there are multiple signals.

Same for `in_distance` etc.